src/quicklist.c

```diff
@@ -1565,6 +1577,8 @@ int quicklistPopCustom(quicklist *quicklist, int where, unsigned char **data,
         return 0;
     }
 
+    assert(node->encoding != QUICKLIST_NODE_ENCODING_LZF);
```

Comment:
I misunderstood this line of code, you want to pop and verify `tail` is not lzf?
If that, we should check it after `quicklistDelIndex` and change it to `assert(_ql->head->encoding != QUICKLIST_NODE_ENCODING_LZF && _ql->tail->encoding != QUICKLIST_NODE_ENCODING_LZF ); `
@oranagra WDYT?

Reply:
On second thought, I think the position should be right now, we just want to avoid inserting data into lzf.
In the sense of oranagra, should we add assertions to `quicklistPush*`?

Reply:
regarding push, you mean to check the status of head and tail, make sure it is not compressed ? why push specific ? there are a lot of commands that will end up inserting keys in the head or in the ket (set, insert etc.)

Reply:
this is what i requested here: https://github.com/redis/redis/pull/9779#discussion_r749598027
i.e. to add an assertion in `push` and `pop`.
AFAIK, `set` and `insert` should already have code to check if it's compressed and do the decompression if necessary.
same for all the code that deletes a node (promotes another node to be `head` or `tail`.
what's "special" about push and pop, is that they assume the node is not compressed, but they don't validate it.
so we can either add an optional decompression there, or an assertion.
i prefer adding an assertion, if we can add a fuzz tester to validate that we have no bugs elsewhere.
note that if we don't add an assertion, it'll still probably explode (at least for `ziplist`, and not for `plain`), so the assertion is just gonna make the failure more clear.